### PR TITLE
engine: simplify pod registration a bit

### PR DESCRIPTION
**Pull request by nicks**
Hello @landism, @maiamcc,

Please review the following commits I made in branch nicks/pod:

e6535c08ea75f0965d5bcf29c81a852f03eeef36 (2020-01-30 17:13:21 -0500)
engine: simplify pod registration a bit

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
moving towards a system where we have Deployments and ReplicaSets in our data model, and so want to separate out the general-purpose resource tracking code from pod-specific stuff

File: internal/engine/pod.go [hunk 1]
```diff
@@ -114,35 +117,30 @@ func matchPodChangeToManifest(state *store.EngineState, action k8swatch.PodChang
...
+	// Only attach a new pod to the runtime state if it's from the current deploy;
+	// if it's from an old deploy/an old Tilt run, we don't want to be checking it
+	// for status etc.
+	if !isCurrentDeploy {
+		return runtime.Pods[podID], false
+	}
+
 	// Case 1: We haven't seen pods for this ancestor yet.
 	ancestorUID := action.AncestorUID
 	isAncestorMatch := ancestorUID != ""
```
Comment (maiamcc) [inline internal/engine/pod.go]: this line has confused me in the past -- I think this variable is more appropriately named `hasAncestor`?

Comment (maiamcc) [inline internal/engine/pod.go]: this line has confused me in the past -- I think this variable is more appropriately named `hasAncestor`?

File: internal/engine/pod.go [hunk 2]
```diff
@@ -114,35 +117,30 @@ func matchPodChangeToManifest(state *store.EngineState, action k8swatch.PodChang
...
-			// for status etc.
-			runtime.PodAncestorUID = ancestorUID
-			runtime.Pods = make(map[k8s.PodID]*store.Pod)
-			runtime.Pods[podID] = podInfo
-			ms.RuntimeState = runtime
+			PodID: podID,
 		}
+
+		runtime.PodAncestorUID = ancestorUID
+		runtime.Pods = make(map[k8s.PodID]*store.Pod)
```
Comment (maiamcc) [inline internal/engine/pod.go]: any reason not to initialize this map in `GetOrCreateK8sRuntimeState`?

File: internal/engine/pod.go [hunk 3]
```diff
@@ -151,19 +149,11 @@ func maybeTrackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*s
...
-			Namespace:  ns,
-			HasSynclet: hasSynclet,
-			SpanID:     runtimelog.SpanIDForPod(podID),
-		}
-		if isCurrentDeploy {
-			// Only attach a new pod to the runtime state if it's from the current deploy;
-			// if it's from an old deploy/an old Tilt run, we don't want to be checking it
-			// for status etc.
-			runtime.Pods[podID] = podInfo
+			PodID: podID,
```
Comment (maiamcc) [inline internal/engine/pod.go]: these two cases are increasingly similar, is there any way to consolidate? (I have an instinct that yes but I'm not seeing exactly how, so 🤷‍♀)

File: internal/engine/pod.go (hunk 1 above)
Comment (nicks) [reply internal/engine/pod.go]: nope, that's not what this field means. There are more comments on it where the PodChangeAction struct is defined. I can change the name of the field if that helps.

File: internal/engine/pod.go (hunk 2 above)
Comment (nicks) [reply internal/engine/pod.go]: we're clearing it, not initializing it. will add a comment on that too

File: internal/engine/pod.go (hunk 3 above)
Comment (nicks) [reply internal/engine/pod.go]: done

**Pull request merged**